net/icmp/icmp_sendmsg.c

```diff
@@ -424,6 +424,8 @@ ssize_t icmp_sendmsg(FAR struct socket *psock, FAR struct msghdr *msg,
         {
           if (ret == -ETIMEDOUT)
             {
+              ret = -EAGAIN;
```

Comment:
Maybe we can move this as `else` for `if (!net_ipv4addr_maskcmp(state.snd_toaddr, dev->d_ipaddr, dev->d_netmask))`?

Reply:
Done.